### ISSUE TITLE
Photo Attachments Not Consistently Sending

### ISSUE THREAD

**Issue by 1ibno** [moezbhatti/qksms#1940]
VERSION(s):
Both 3.10.1 and 3.9.4
(from f-droid)

HARDWARE:
Pixel 4A (rooted)
TQ1A.221205.011

DESCRIPTION:
App seems unable to consistently allow the sending of image attachments (specifically: photos).

STEPS TO REPRODUCE:
1. Compose (or reply to) new message
2. Select 'Attach a photo' -OR- 'Take a photo'
3. Take either action
4. Attempt to send 
5. Receive error: 'Failed to send. Tap to try again'. 

EXPECTED BEHAVIOR:
Message should send with attached image.

OBSERVED BEHAVIOR:
Some messages (with their attached images) are not successfully sent—while others are.

![1](https://user-images.githubusercontent.com/94773681/227719786-1fd827b0-880a-4cb8-9944-de094fa2fe94.png)
![2](https://user-images.githubusercontent.com/94773681/227719788-37440660-f1d7-4b24-ae1d-edd574632e8d.png)

**Comment by vdbhb59** [comment]
Same here. For me it hardly goes ever.

**Comment by sirfredrick** [comment]
I've found a workaround. Changing "Auto-compress MMS image attachments" to either 300KB or 600KB allows the message to be sent but can reduce image quality.

**Comment by tcecyk** [comment]
> I've found a workaround. Changing "Auto-compress MMS image attachments" to either 300KB or 600KB ..

600KB was recommended before: https://github.com/moezbhatti/qksms/issues/1569#issuecomment-1022886068

adding a helptext suggestion to lower the limit when the try-again occurs repeatedly on an attachment is a conservative and helfpul change

**Comment by sethro2** [comment]
We had been using the 600 KB workaround on all our phones thru Verizon at the time.  Upon switching to Mint Mobile (T-Mobile) we had not seen issues with using "Automatic" for compression initially.  I would say within a day this stops working for whatever reason... any attempt to send pictures to groups or individuals fails consistently regardless of size.

Switching back to 600 KB did not help which was odd.  I verified the APN configs and also tried a few tweaks.  The only fix for now is "force closing" and reopening the app.  Pictures send flawlessly thereafter until the issue reoccurs.  Very strange stuff.

Device:  OnePlus 9 Pro
OS:  /e/OS S (Android 12) and T (Android 13)
App Version:  v3.9.4 & v3.10.1

**Comment by vdbhb59** [comment]
Nope, for my carrier it works only with stupid google message app for whatever reason. All API or carrier tweaks do squat to the QKSMS. So frustrating.

**Comment by sethro2** [comment]
Well gang, I found some forum posts about there being issues when WiFi Calling is enabled and using texting apps other than Google Messenger. 

For example:  https://www.reddit.com/r/textra/comments/a4hkvg/tmobile_mms/

I am testing the following and will advise if this stabilizes MMS functionality:

WiFi Calling Disabled
Compression set to 600 kb

**Comment by sethro2** [comment]
No dice.  Pics began to fail to send within the hour.  Ugh...
